### PR TITLE
Add intro for "Distributed SQL over TiKV"

### PR DESCRIPTION
**Pull request by breezewish**

### CODE REVIEW

File: src/distributed-sql/introduction.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Distributed SQL over TiKV
+
+TiKV is served as the storage layer for [TiDB], a distributed HTAP SQL database.
```
Comment (Hoverbear) [inline src/distributed-sql/introduction.md]:
```suggestion
TiKV serves as the storage layer for [TiDB], a distributed HTAP SQL database.
```

File: src/distributed-sql/introduction.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+So far, we have only explained how a distributed and transactional Key-Value
```
Comment (Hoverbear) [inline src/distributed-sql/introduction.md]:
```suggestion
So far, we have only explained how a distributed transactional Key-Value
```

File: src/distributed-sql/introduction.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+database is implemented. However this is still far from serving a SQL database.
+We will explore and cover the following things in this chapter:
+
+- Storing
+
+  In this section we will see how TiDB relational structure (i.e. SQL table
+  records and indexes) are encoded into the Key-Value form in latest version.
+  We will also explore a new Key-Value format that is going to be implemented
+  recently and some outlook for some even better Key-Value formats in future.
```
Comment (Hoverbear) [inline src/distributed-sql/introduction.md]:
```suggestion
  soon and some insights on even better Key-Value formats in future.
```

File: src/distributed-sql/introduction.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+- Distributed SQL (DistSQL)
+
+  Storing data distributedly in TiKV only utilizes distributed IO resources,
```
Comment (Hoverbear) [inline src/distributed-sql/introduction.md]:
```suggestion
  Storing data in a distributed manner using TiKV only utilizes distributed I/O resources,
```

File: src/distributed-sql/introduction.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  while the TiDB node that receives SQL query is still in charge of processing
+  all rows. What if we step further, delegating some processing tasks into TiKV
```
Comment (Hoverbear) [inline src/distributed-sql/introduction.md]:
```suggestion
  all rows. We can go a step further by delegating some processing tasks into TiKV
```

File: src/distributed-sql/introduction.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  nodes? In this way, we can utilize distributed CPU resources! In this section,
```
Comment (Hoverbear) [inline src/distributed-sql/introduction.md]:
```suggestion
  nodes. This way, we can utilize distributed CPU resources! In this section,
```

File: src/distributed-sql/introduction.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  we will take a look at these supported physical plan executors so far in TiKV
+  and see how they enable TiDB executing SQL queries in a distributed way.
+
+- TiKV Query Execution Engine
+
+  When talking about executors we cannot ignore discussing about the execution
```
Comment (Hoverbear) [inline src/distributed-sql/introduction.md]:
```suggestion
  When talking about executors we cannot ignore discussing the execution
```

File: src/distributed-sql/introduction.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  engine. Although executors running on TiKV are highly simplified and limited,
+  we still need to carefully design the execution engine which is critical to
+  the performance. This section will cover the traditional Volcano model
+  execution engine used before TiKV 3.0, for example, how it works, pros and
+  cons and the architecture.
```
Comment (Hoverbear) [inline src/distributed-sql/introduction.md]:
```suggestion
  cons, and the architecture.
```

File: src/distributed-sql/introduction.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
+# Distributed SQL over TiKV
+
+TiKV serves as the storage layer for [TiDB], a distributed HTAP SQL database.
+So far, we have only explained how a distributed transactional Key-Value
+database is implemented. However this is still far from serving a SQL database.
+We will explore and cover the following things in this chapter:
+
+- Storing
+
+  In this section we will see how TiDB relational structure (i.e. SQL table
```
Comment (Hoverbear) [inline src/distributed-sql/introduction.md]:
```suggestion
  In this section we will see how the TiDB relational structure (i.e. SQL table
```

File: src/distributed-sql/introduction.md [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  tasks into TiKV nodes. This way, we can utilize distributed CPU resources! In
+  this section, we will take a look at these supported physical plan executors
+  so far in TiKV and see how they enable TiDB executing SQL queries in a
+  distributed way.
+
+- TiKV Query Execution Engine
+
+  When talking about executors we cannot ignore discussing the execution engine.
+  Although executors running on TiKV are highly simplified and limited, we still
+  need to carefully design the execution engine which is critical to the
```
Comment (Hoverbear) [inline src/distributed-sql/introduction.md]:
```suggestion
  need to carefully design the execution engine. It is critical to the
```

File: src/distributed-sql/introduction.md [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  performance. This section will cover the traditional Volcano model execution
```
Comment (Hoverbear) [inline src/distributed-sql/introduction.md]:
```suggestion
  performance of the system. This section will cover the traditional Volcano model execution
```

File: src/distributed-sql/introduction.md [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+  engine used before TiKV 3.0, for example, how it works, pros and cons, and the
+  architecture.
+
+- Vectorization
+
+  Vectorization is a technique that performs computing over a batch of values.
+  By introducing vectorization into the exeuction engine, we will achieve higher
```
Comment (Hoverbear) [inline src/distributed-sql/introduction.md]:
```suggestion
  By introducing vectorization into the execution engine, we will achieve higher
```

**Review by Hoverbear — APPROVED** [review]
Rest LGTM

File: src/distributed-sql/introduction.md [hunk 13]
```diff
@@ -0,0 +1,41 @@
+# Distributed SQL over TiKV
+
+TiKV serves as the storage layer for [TiDB], a distributed HTAP SQL database.
```
Comment (huachaohuang) [inline src/distributed-sql/introduction.md]:
```suggestion
TiKV is the storage layer for [TiDB], a distributed HTAP SQL database.
```

File: src/distributed-sql/introduction.md [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+So far, we have only explained how a distributed transactional Key-Value
+database is implemented. However this is still far from serving a SQL database.
+We will explore and cover the following things in this chapter:
+
+- Storing
```
Comment (huachaohuang) [inline src/distributed-sql/introduction.md]:
```suggestion
- Storage
```

File: src/distributed-sql/introduction.md [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+
+  In this section we will see how TiDB relational structure (i.e. SQL table
+  records and indexes) are encoded into the Key-Value form in latest version.
```
Comment (huachaohuang) [inline src/distributed-sql/introduction.md]:
```suggestion
  records and indexes) are encoded into the Key-Value form in the latest version.
```

**Review by huachaohuang — COMMENTED** [review]
Rest LGTM

**Comment by breezewish** [comment]
@huachaohuang @Hoverbear PTAL again. Thanks!

**Review by huachaohuang — APPROVED** [review]
LGTM

**Pull request merged**